### PR TITLE
feature: add check for non existent member

### PR DESCRIPTION
**Pull request by ShenDing1125**
Add checks to prevent errors during member transfer.

### CODE REVIEW

**Comment by ShenDing1125** [comment]
[【名單撥打】無限業務把單轉到學米，需將名單改回至無限帳號上](https://trello.com/c/5zDbFKlu/5066-%E3%80%90%E5%90%8D%E5%96%AE%E6%92%A5%E6%89%93%E3%80%91%E7%84%A1%E9%99%90%E6%A5%AD%E5%8B%99%E6%8A%8A%E5%96%AE%E8%BD%89%E5%88%B0%E5%AD%B8%E7%B1%B3%EF%BC%8C%E9%9C%80%E5%B0%87%E5%90%8D%E5%96%AE%E6%94%B9%E5%9B%9E%E8%87%B3%E7%84%A1%E9%99%90%E5%B8%B3%E8%99%9F%E4%B8%8A)

File: src/components/sale/SalesLeadTable.tsx [hunk 1]
```diff
@@ -960,4 +1014,22 @@ const TRANSFER_LEADS = gql`
   }
 `
 
+const GET_MEMBER_APPID = gql`
```
Comment (Ponchimeow) [inline src/components/sale/SalesLeadTable.tsx:1017]: CamelCase

`GetMemberAppID`

File: src/components/sale/SalesLeadTable.tsx [hunk 2]
```diff
@@ -960,4 +1014,22 @@ const TRANSFER_LEADS = gql`
...
+  query GET_MEMBER_APPID($managerId: String!) {
```
Comment (Ponchimeow) [inline src/components/sale/SalesLeadTable.tsx:1018]: CamelCase

`GetMemberAppID`

File: src/components/sale/SalesLeadTable.tsx [hunk 3]
```diff
@@ -960,4 +1014,22 @@ const TRANSFER_LEADS = gql`
...
+    member(where: { id: { _eq: $managerId } }) {
+      app_id
+    }
+  }
+`
+
+const GET_MANAGER_APPID = gql`
```
Comment (Ponchimeow) [inline src/components/sale/SalesLeadTable.tsx:1025]: CamelCase

`GetManagerAppID`

the result is list
could be GetManagerAppIDCollection or GetManagerAppIDList

File: src/components/sale/SalesLeadTable.tsx [hunk 4]
```diff
@@ -960,4 +1014,22 @@ const TRANSFER_LEADS = gql`
...
+  query GET_MANAGER_APPID($managerIds: [String]!) {
```
Comment (Ponchimeow) [inline src/components/sale/SalesLeadTable.tsx:1026]: CamelCase

`GetManagerAppID`

the result is list
could be GetManagerAppIDCollection or GetManagerAppIDList

File: src/components/sale/SalesLeadTable.tsx (hunk 1 above)
Comment (ShenDing1125) [reply src/components/sale/SalesLeadTable.tsx]: Thanks for the advice!

File: src/components/sale/SalesLeadTable.tsx [hunk 5]
```diff
@@ -501,6 +501,82 @@ const SalesLeadTable: React.VFC<{
...
+    const nonMatchingAppIds: Record<string, string>[] = []
+
+    businessAppId.current = memberAppId?.member[0]?.app_id
+    if (businessAppId.current == null) {
+      alert(`此會員不存在: ${managerId}`)
+      setManagerId('')
+      return
+    }
+
+    for (const _managerId of managerAppIds.member) {
```
Comment (Ponchimeow) [inline src/components/sale/SalesLeadTable.tsx:534]: Why use an underscore as the parameter prefix?

Comment (ShenDing1125) [reply src/components/sale/SalesLeadTable.tsx]: Thanks for the suggestion! I've removed the underscores to make the code more readable. Appreciate your feedback!

File: src/components/sale/SalesLeadTable.tsx (hunk 2 above)
Comment (kkshyu) [reply src/components/sale/SalesLeadTable.tsx]:
```
const UpperCamelCase = gql`
    mutation UpperCamelCase {
    }
`
// setter, getter -> function -> lowerCamelCase(...)
const [lowerCamelCase, {data, loading, error} ] = useMutation(UpperCamelCase)
```

File: src/components/sale/SalesLeadTable.tsx (hunk 5 above)
Comment (kkshyu) [reply src/components/sale/SalesLeadTable.tsx]: there are different types of naming convention
```
_variable
__variable
variable_
variable__
_variable_
__variable__
```

**Pull request merged**